pkg/strategy/ewo_dgtrd/strategy.go

```diff
@@ -0,0 +1,409 @@
+	if !ok {
+		log.Errorf("cannot get indicatorSet of %s", s.Symbol)
+		return nil
+	}
+	orderbook, ok := session.OrderStore(s.Symbol)
+	if !ok {
+		log.Errorf("cannot get orderbook of %s", s.Symbol)
+		return nil
+	}
+	position, ok := session.Position(s.Symbol)
+	if !ok {
+		log.Errorf("cannot get position of %s", s.Symbol)
+		return nil
+	}
+	s.tradeCollector = bbgo.NewTradeCollector(s.Symbol, position, orderbook)
+	s.tradeCollector.OnTrade(func(trade types.Trade, profit, netprofit fixedpoint.Value) {
+		if !profit.IsZero() {
+			log.Warnf("generate profit: %v, netprofit: %v, trade: %v", profit, netprofit, trade)
+		}
+	})
+
+	s.tradeCollector.OnPositionUpdate(func(position *types.Position) {
+		log.Infof("position changed: %s", position)
+		tp = position.Type()
+		if tp == types.PositionLong {
+			buyPrice = position.AverageCost
+		} else if tp == types.PositionShort {
+			sellPrice = position.AverageCost
+		} else {
+			buyPrice = fixedpoint.Zero
+			sellPrice = fixedpoint.Zero
+		}
+	})
+	s.tradeCollector.BindStream(session.UserDataStream)
+
+	//s.SmartStops.RunStopControllers(ctx, session, s.tradeCollector)
+
+	/*store, ok := session.MarketDataStore(s.Symbol)
+	if !ok {
+		log.Errorf("cannot get marketdatastore of %s", s.Symbol)
+		return nil
+	}*/
+
+	/*window, ok := store.KLinesOfInterval(s.Interval)
+	if !ok {
+		log.Errorf("cannot get klinewindow of %s", s.Interval)
+	}*/
+	var ma5, ma34, ma50, ewo types.Series
+	if s.UseEma {
+		ma5 = indicatorSet.EWMA(types.IntervalWindow{s.Interval, 5})
+		ma34 = indicatorSet.EWMA(types.IntervalWindow{s.Interval, 34})
+		ma50 = indicatorSet.EWMA(types.IntervalWindow{s.Interval, 50})
+	} else {
+		ma5 = indicatorSet.SMA(types.IntervalWindow{s.Interval, 5})
+		ma34 = indicatorSet.SMA(types.IntervalWindow{s.Interval, 34})
+		ma50 = indicatorSet.SMA(types.IntervalWindow{s.Interval, 50})
+	}
+	ewo = types.Mul(types.Minus(types.Div(ma5, ma34), 1.0), 100.)
+	var ewoSignal EwoSignal
+	if s.UseEma {
+		ewoSignal = &indicator.EWMA{IntervalWindow: types.IntervalWindow{s.Interval, s.SignalWindow}}
+	} else {
+		ewoSignal = &indicator.SMA{IntervalWindow: types.IntervalWindow{s.Interval, s.SignalWindow}}
+	}
+	session.MarketDataStream.OnKLineClosed(func(kline types.KLine) {
+		if kline.Symbol != s.Symbol {
+			return
+		}
+
+		lastPrice, ok := session.LastPrice(s.Symbol)
+		if !ok {
+			log.Errorf("cannot get last price")
+			return
+		}
+
+		// cancel non-traded orders
+		var toCancel []types.Order
+		var toRepost []types.SubmitOrder
+		for _, order := range orderbook.Orders() {
+			if order.Status == types.OrderStatusNew || order.Status == types.OrderStatusPartiallyFilled {
+				toCancel = append(toCancel, order)
+			}
+		}
+		if len(toCancel) > 0 {
+			if err := orderExecutor.CancelOrders(ctx, toCancel...); err != nil {
+				log.WithError(err).Errorf("cancel order error")
+			}
+
+			s.tradeCollector.Process()
+		}
+
+		// well, only track prices on 1m
+		if kline.Interval == types.Interval1m {
```

Comment:
invert this if ?

if != then return ?

Reply:
condition as follows:
1. if interval == 1m, cancel and repost the orders to the latest low, also check the stoploss or trailingstop. return without further action.
2. if interval != 1m (== s.Interval), cancel all and track indicators for further decisions

line 313 will do the early return for case 1 but not in this area.